src/SubscriptionService/SubscriptionService.cs

```diff
@@ -0,0 +1,492 @@
+﻿namespace SubscriptionService
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Configuration;
+    using EventStore.ClientAPI;
+    using EventStore.ClientAPI.SystemData;
+
+    /// <summary>
+    /// </summary>
+    /// <seealso cref="SubscriptionService.ISubscriptionService" />
+    /// <seealso cref="ISubscriptionService" />
+    public class SubscriptionService : ISubscriptionService
+    {
+        #region Fields
+
+        /// <summary>
+        /// The connection string
+        /// </summary>
+        private readonly String ConnectionString;
+
+        /// <summary>
+        /// The destroy lock
+        /// </summary>
+        private readonly Object DestroyLock = new Object();
+
+        /// <summary>
+        /// The event store connection
+        /// </summary>
+        private readonly IEventStoreConnection EventStoreConnection;
+
+        /// <summary>
+        /// The HTTP client
+        /// </summary>
+        private readonly HttpClient HttpClient;
+
+        /// <summary>
+        /// The lock object
+        /// </summary>
+        private readonly Object LockObject = new Object();
+
+        /// <summary>
+        /// The manual reset event
+        /// </summary>
+        private readonly ManualResetEvent ManualResetEvent = new ManualResetEvent(true);
+
+        /// <summary>
+        /// The subscriptions
+        /// </summary>
+        private readonly List<Subscription> Subscriptions;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SubscriptionService" /> class.
+        /// </summary>
+        /// <param name="subscriptions">The subscriptions.</param>
+        /// <param name="eventStoreConnection">The event store connection.</param>
+        /// <param name="username">The username.</param>
+        /// <param name="password">The password.</param>
+        public SubscriptionService(List<Subscription> subscriptions,
+                                   IEventStoreConnection eventStoreConnection,
+                                   String username = "admin",
```

Comment:
Could the user not pass in UserCredentials?

Reply:
They are required in CreatePersistentSubscriptionAsync.